dlt/destinations/impl/clickhouse/clickhouse.py

```diff
@@ -0,0 +1,416 @@
+            r"Decimal\((?P<precision>\d+)\s*(?:,\s*(?P<scale>\d+))?\)", db_type
+        ):
+            precision, scale = decimal_match.groups()  # type: ignore[assignment]
+            precision = int(precision)
+            scale = int(scale) if scale else 0
+            db_type = "Decimal"
+
+        if db_type == "Decimal" and (precision, scale) == self.capabilities.wei_precision:
+            return dict(data_type="wei")
+
+        return super().from_db_type(db_type, precision, scale)
+
+
+class ClickHouseLoadJob(LoadJob, FollowupJob):
+    def __init__(
+        self,
+        file_path: str,
+        table_name: str,
+        client: ClickHouseSqlClient,
+        staging_credentials: Optional[CredentialsConfiguration] = None,
+    ) -> None:
+        file_name = FileStorage.get_file_name_from_file_path(file_path)
+        super().__init__(file_name)
+
+        qualified_table_name = client.make_qualified_table_name(table_name)
+
+        bucket_path: str = (
+            NewReferenceJob.resolve_reference(file_path)
+            if NewReferenceJob.is_reference_job(file_path)
+            else ""
+        )
+        file_name = (
+            FileStorage.get_file_name_from_file_path(bucket_path) if bucket_path else file_name
+        )
+        file_extension = os.path.splitext(file_name)[1][
+            1:
+        ].lower()  # Remove dot (.) from file extension.
+
+        if file_extension not in ["parquet", "jsonl"]:
+            raise LoadJobTerminalException(
+                file_path, "ClickHouse loader Only supports parquet and jsonl files."
+            )
+
+        bucket_url = urlparse(bucket_path)
+        bucket_scheme = bucket_url.scheme
+
+        file_extension = cast(SUPPORTED_FILE_FORMATS, file_extension)
+        clickhouse_format: str = FILE_FORMAT_TO_TABLE_FUNCTION_MAPPING[file_extension]
+        if file_extension == "parquet":
+            # Auto works for parquet
+            compression = "auto"
+        else:
+            # It does not work for json
+            compression = "none" if config.get("data_writer.disable_compression") else "gz"
+
+        statement: str = ""
+
+        if bucket_scheme in ("s3", "gs", "gcs"):
+            bucket_http_url = convert_storage_to_http_scheme(bucket_url)
+
+            if isinstance(staging_credentials, AwsCredentialsWithoutDefaults):
+                access_key_id = staging_credentials.aws_access_key_id
+                secret_access_key = staging_credentials.aws_secret_access_key
+            elif isinstance(staging_credentials, GcpCredentials):
+                access_key_id = dlt.config["destination.filesystem.credentials.gcp_access_key_id"]
+                secret_access_key = dlt.config[
+                    "destination.filesystem.credentials.gcp_secret_access_key"
+                ]
+            else:
+                access_key_id = None
+                secret_access_key = None
+
+            structure = "auto"
+
+            template = Template("""
```

Comment:
why do you use jinja template to render this? do we have it in dependencies? is it maybe coming with clickhouse client? if not then most probably this will stop working on clean install

Reply:
There was no real reason to use jinja templating in this simple case, agreed! It makes the code less readable now that I had another look.

Removed it in favor of built in string formatting.